wolfcrypt/src/pkcs7.c

```diff
@@ -10253,10 +10318,9 @@ WOLFSSL_API int wc_PKCS7_DecodeEnvelopedData(PKCS7* pkcs7, byte* in,
             if ((ret = wc_PKCS7_StreamEndCase(pkcs7, &tmpIdx, &idx)) != 0) {
                 break;
             }
-            pkcs7->stream->expected = encryptedContentSz;
+            pkcs7->stream->expected = 0;
```

Comment:
Expected size for next read should be encrypted content total size and then after making sure all content data is there it can go through checking for octet strings of encrypted content

Reply:
Thanks, adjusted to bump content processing into the next switch statement.  This change also added one additional variable slot to the StreamStoreVar() / StreamGetVar() APIs.